src/components/DepositWidget/Form.tsx

```diff
@@ -85,11 +78,9 @@ export const Form: React.FC<FormProps> = (props: FormProps) => {
     if (!error) {
       setLoading(true)
       props.onSubmit(parseAmount(amountInput, decimals)).then(() => {
-        if (mounted.current) {
-          setLoading(false)
-          setValidatorActive(false)
-          cancelForm()
-        }
+        setLoading(false)
+        setValidatorActive(false)
+        cancelForm()
```

Comment:
The new hook doesn't cover the side effects of this function, or does it?
`props.onClose()` will now be called even when unmounted.

Reply:
Yes, but I think in this case it's ok, since it triggers `setForm` in `Row` . It will only be a problem if a different form was opened and would force close as a result.

Maybe bring in also the `useIfMounted` hook for such cases.